### PR TITLE
Add support for Windows to Azure image publishing pipeline

### PR DESCRIPTION
**Pull request by jsturtevant**
Updates the Azure publishing scripts to support multiple Operating System and Versions

/assign @CecileRobertMichon 

In process of testing the changes to the scripts
/hold

### CODE REVIEW

**Comment by jsturtevant** [comment]
quick update, working on creating a new Offer in our market place images which is taking a bit longer than expected

**Comment by jsturtevant** [comment]
/hold cancel

I've had a successful publish of the images. 

Note I integrated the disk-version publish into the stages instead of using AzDo releases

**Comment by jsturtevant** [comment]
The images are published at: `az vm image list --publisher cncf-upstream --offer capi-windows -o table --all`

File: images/capi/packer/azure/.pipelines/stages.yaml [hunk 1]
```diff
@@ -34,3 +36,12 @@ stages:
...
       AZURE_CLIENT_SECRET: $(AZURE_CLIENT_SECRET_SKU)
+
+  - stage: disk_version
+    condition: and(succeeded(), eq(variables.CLEANUP, 'False'))
+    jobs:
+    - template: create-disk-version.yaml
+    variables:
+      AZURE_TENANT_ID: $(AZURE_TENANT_ID_SKU)
+      AZURE_CLIENT_ID: $(AZURE_CLIENT_ID_SKU)
+      AZURE_CLIENT_SECRET: $(AZURE_CLIENT_SECRET_SKU)
```
Comment (CecileRobertMichon) [inline images/capi/packer/azure/.pipelines/stages.yaml:47]: nit: missing newline

File: images/capi/packer/azure/scripts/new-disk-version.sh [hunk 2]
```diff
@@ -1,12 +1,18 @@
 #!/bin/bash -e
 
-cd '/build' || exit
 echo "PWD: $PWD"
+ls sku/
```
Comment (CecileRobertMichon) [inline images/capi/packer/azure/scripts/new-disk-version.sh:4]: are these debug statements?

File: images/capi/packer/azure/.pipelines/create-sku.yaml [hunk 3]
```diff
@@ -8,6 +8,8 @@
 # - PUBLISHER - the name of the publisher to create the sku for
 # - OFFER - the name of the offer to create the sku for
 # - SKU_TEMPLATE_FILE - the base template file to use for the sku
+# - OS - target of build e.g. `Ubuntu/Windows`
+# - OS_VERSION - target of build e.g. `1809/2004/2019`
```
Comment (CecileRobertMichon) [inline images/capi/packer/azure/.pipelines/create-sku.yaml:12]: let's add 18.04 as an example. for `os` your removing the periods so it won't matter but in the new disk version script it's being used as-is for label and description

**Review by CecileRobertMichon — COMMENTED** [review]
lgtm, minor comments

File: images/capi/packer/azure/.pipelines/create-sku.yaml (hunk 3 above)
Comment (jsturtevant) [reply images/capi/packer/azure/.pipelines/create-sku.yaml:12]: meant to do the 18.04 but got that mixed up with windows version :-)

File: images/capi/packer/azure/scripts/new-disk-version.sh (hunk 2 above)
Comment (jsturtevant) [reply images/capi/packer/azure/scripts/new-disk-version.sh:4]: yes, will remove them

File: images/capi/packer/azure/.pipelines/create-disk-version.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Required pipeline variables:
+# - BUILD_POOL - Azure DevOps build pool to use
+# - CONTAINER_IMAGE - Dev container image URL to use. Should have Azure CLI, Packer and Ansible.
+# - AZURE_TENANT_ID - tenant ID
+# - AZURE_CLIENT_ID - Service principal ID
+# - AZURE_CLIENT_SECRET - Service principal secret
+# - OS - target of build e.g. `Ubuntu/Windows`
+# - OS_VERSION - target of build e.g. `1809/2004/2019`
```
Comment (CecileRobertMichon) [inline images/capi/packer/azure/.pipelines/create-disk-version.yaml:8]:
```suggestion
# - OS_VERSION - target of build e.g. `18.04/2004/2019`

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**